test/common/upstream/health_check_fuzz.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package test.common.upstream;
+
+import "validate/validate.proto";
+import "test/fuzz/common.proto";
+import "envoy/config/core/v3/health_check.proto";
+
+message HttpRespond { //TODO: Split this across headers, body, and trailers?
+  test.fuzz.Headers headers = 1;
+  //https://github.com/nodejs/http-parser/blob/master/http_parser.c#L852
+  string status = 2 [(validate.rules).string.pattern = "[0-9]+$*"];
+}
+
+message Respond {
+  HttpRespond http_respond = 1;
+  //TODO: TCP and gRPC Respond
+  bool second_host = 4;
+}
+
+message StreamCreate {
+  bool second_host = 1;
+}
+
+message AdvanceTime {
+  int32 ms_advanced = 1 [(validate.rules).int32 = {gt: 0 lt: 3600000}]; //Capped at one hour
+}
+
+message Action {
+  oneof action_selector {
+    Respond respond = 1;
+    StreamCreate stream_create = 2;
+    //TODO: respondBody, respondTrailers
+    AdvanceTime advance_time = 3;
+  }
+}
+
+message HealthCheckTestCase {
+  envoy.config.core.v3.HealthCheck health_check_config = 1
+      [(validate.rules).message.required = true];
+  repeated Action actions = 2;
+  bool create_second_host = 3;
+  bool http_verify_cluster = 4; //Determines if verify cluster setting is on
```

Comment:
another suggestion for an initial setup, i noticed some test cases set health flags before the start (SuccessStartFailedSuccessFirst) to test beheavior for hosts that never become healthy https://github.com/envoyproxy/envoy/pull/4965.